### DISCUSSION TITLE
Where is the documentation for .ignore format?

### DISCUSSION THREAD

**Discussion by dbarnett** [BurntSushi/ripgrep#2895] [Q&A] [answered]
Is there an official spec for what goes in a .ignore file? I like the spirit of it but I can't seem to find a user-oriented explanation of how it's supposed to work.

Starting from https://crates.io/crates/ignore and https://docs.rs/ignore/, I'm not seeing any explanation of what goes in .ignore files besides the description hinting that it relates to .gitignore files.

**Comment by BurntSushi** [comment, marked as answer]
It's [mentioned in the guide](https://github.com/BurntSushi/ripgrep/blob/master/GUIDE.md#automatic-filtering):

> For a more in depth description of how glob patterns in a `.gitignore` file are interpreted, please see `man gitignore`.

I suppose it's not fully precise in that it says `.gitignore` and not `.ignore`. But `.ignore` has the same semantics as `.gitignore`. The only reason it's in a distinct file is for [precedence reasons](https://docs.rs/ignore/latest/ignore/struct.WalkBuilder.html#ignore-rules).

Also, the `ignore` crate API and documentation are in general severely lacking. It's effectively a proof of concept that works decently enough.

**Reply by dbarnett** [reply]
Thanks! That does give me some starting points for the answers, but when considering integrating it with other projects (example: https://github.com/martinvonz/jj/issues/3525), it'd be pretty awkward to point users deep into ripgrep docs for answers on how to use it.

**Reply by BurntSushi** [reply]
I agree that the current state is not great or ideal. Improving `ignore`, and its docs, has been on my TODO list for years. I do hope I'll get to it some day, but it probably isn't happening any time soon.
